### DISCUSSION TITLE
Enable logs in Docker output

### DISCUSSION THREAD

**Discussion by bmingles** [pocketbase/pocketbase#5121] [Q&A] [answered]
I'm trying to figure out how to enable logging to the terminal when running Pocketbase in a Docker container. I am using the following Docker file:

```Dockerfile
# Pocketbase server
FROM golang:1.22.4

WORKDIR /pb

COPY go.mod go.sum ./
RUN go mod download

COPY *.go ./

# Build
RUN CGO_ENABLED=0 GOOS=linux go build -o /pb/myapp

COPY ./pb_migrations /pb/pb_migrations

COPY ./pb_hooks /pb/pb_hooks

# Optional:
# To bind to a TCP port, runtime parameters must be supplied to the docker command.
# But we can document in the Dockerfile what ports
# the application is going to listen on by default.
# https://docs.docker.com/engine/reference/builder/#expose
EXPOSE 8080

# Run
CMD ["/pb/myapp", "serve", "--http=0.0.0.0:8080"]
```

and the following `main.go` file:

```go
package main

import (
	"log"
	"os"

	"github.com/labstack/echo/v5"
	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/apis"
	"github.com/pocketbase/pocketbase/core"
	"github.com/pocketbase/pocketbase/plugins/jsvm"
	"github.com/pocketbase/pocketbase/plugins/migratecmd"
)

func main() {
    app := pocketbase.New()

    var hooksDir string
	app.RootCmd.PersistentFlags().StringVar(
		&hooksDir,
		"hooksDir",
		"",
		"the directory with the JS app hooks",
	)

	var hooksWatch bool
	app.RootCmd.PersistentFlags().BoolVar(
		&hooksWatch,
		"hooksWatch",
		true,
		"auto restart the app on pb_hooks file change",
	)

    var migrationsDir string
	app.RootCmd.PersistentFlags().StringVar(
		&migrationsDir,
		"migrationsDir",
		"",
		"the directory with the user defined migrations",
	)

    var automigrate bool
	app.RootCmd.PersistentFlags().BoolVar(
		&automigrate,
		"automigrate",
		true,
		"enable/disable auto migrations",
	)

    // load jsvm (hooks and migrations)
	jsvm.MustRegister(app, jsvm.Config{
		MigrationsDir: migrationsDir,
	})

    // migrate command (with js templates)
	migratecmd.MustRegister(app, app.RootCmd, migratecmd.Config{
		TemplateLang: migratecmd.TemplateLangJS,
		Automigrate:  automigrate,
		Dir:          migrationsDir,
	})

    // serves static files from the provided public dir (if exists)
    app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
        e.Router.GET("/", func(c echo.Context) error {
            return c.String(404, "Not Found")
        })

        e.Router.GET("/*", apis.StaticDirectoryHandler(os.DirFS("./pb_public"), false))

        return nil
    })

    if err := app.Start(); err != nil {
        log.Fatal(err)
    }
}
```

When I run the container, it works fine, but the only output I see is:
<img width="503" alt="image" src="https://github.com/pocketbase/pocketbase/assets/1900643/4a05e3bc-21c7-4149-a52f-2cf2a6b45ca7">

I'm trying to debug an issue where the server stops working and I have to restart the container, but I have no idea how to inspect error logs.

**Comment by ganigeorgiev** [comment, marked as answer]
You can specify the `--dev` flag to the `CMD` instruction.

**Reply by ganigeorgiev** [reply]
> server stops working

It is not clear what do you mean by this, but note that if there was a Go panic you should see it even without the `--dev` flag in the stderr. Double check your custom hooks code, especially if you have transactions since the most common error in this case is usually using the `app.Dao()` instead the `txDao` argument of the transaction which could result in deadlock when used with write queries since the write/transaction db pool is limited to only 1 active connection.

**Reply by bmingles** [reply]
Looks like my issue is that I had tried `--debug` instead of `--dev`. 

Regarding my comment of `server stops working`, this meant that the server stopped allowing me to login. Things are still up and running, but I can no longer interact with the server without restarting it. The only thing I have in my `pb_hooks` directory are 2 util scripts I use with migrations, so shouldn't be anything here that would impact things once it is up and running.

I'll keep things running in `--dev` mode and maybe if this scenario happens again, there will be something meaningful in the logs to help hunt it down

**Comment by psychowood** [comment]
> I'm trying to figure out how to enable logging to the terminal when running Pocketbase in a Docker container. I am using the following Docker file:
> 
> ```dockerfile
> # Pocketbase server
> FROM golang:1.22.4
> 
> WORKDIR /pb
> 
> COPY go.mod go.sum ./
> RUN go mod download
> 
> COPY *.go ./
> 
> # Build
> RUN CGO_ENABLED=0 GOOS=linux go build -o /pb/myapp
> 
> COPY ./pb_migrations /pb/pb_migrations
> 
> COPY ./pb_hooks /pb/pb_hooks
> 
> # Optional:
> # To bind to a TCP port, runtime parameters must be supplied to the docker command.
> # But we can document in the Dockerfile what ports
> # the application is going to listen on by default.
> # https://docs.docker.com/engine/reference/builder/#expose
> EXPOSE 8080
> 
> # Run
> CMD ["/pb/myapp", "serve", "--http=0.0.0.0:8080"]
> ```

Not directly related to your issue, but you may want to add a [VOLUME](https://stackoverflow.com/questions/41935435/understanding-volume-instruction-in-dockerfile) command to remap your `/pb/pb_data` otherwise you'll lose everything when recreating the container. I'd also remap `pb_migrations` and `pb_hooks` too instead of copying.

**Reply by bmingles** [reply]
@psychowood try adding "--dev" to the CMD args of your Docker file
